service/s3.cc

```diff
@@ -2761,11 +2723,7 @@ std::shared_ptr<S3Api> getS3ApiForUri(const std::string & uri)
         accessKey = getAwsAccessKey(accessKeyId);
     }
 
-    auto api = make_shared<S3Api>(accessKeyId, accessKey);
```

Comment:
That's actually one of the problems I am trying to solve: our check consists in attempting to perform a GET on /[bucketname]/ and that operation is sometimes not even allowed. Therefore, from a logical standpoint, it cannot be used as reliable checks on the validity of a buffer. I guess we could refine the check and differentiate between a 404 and a 403, but even there I am not sure that would give us a reliable answer: what if you get a 403 on a bucket that does not exist anyway.